python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -173,7 +174,11 @@ def __init__(
             _preset_defs, "preset_defs", of_type=PresetDefinition
         )
 
-        did_user_provide_resources = bool(resource_defs)
+        did_user_provide_resources = (
```

Comment:
maybe `does_provide_resources` - feels weird to use the term `user` in the code itself